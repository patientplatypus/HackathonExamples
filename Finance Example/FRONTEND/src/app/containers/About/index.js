import React, {Component} from 'react';
import {
  Icon,
  Card,
  Col,
  Row,
  Badge,
  Menu,
  Dropdown,
  Button,
  Modal,
  message
} from 'antd';
// import {cardStyles, vmCard} from '../../../style/MainStyles.js';
import {Link} from "react-router-dom";
import { connect } from 'react-redux'
import './local.scss'

class About extends Component {
  constructor(props) {
    super(props);
    this.state = {
    }
  }



  render() {
      return (
        <div>
          <h1 className="whitecolor">
            ABOUT
          </h1>
        </div>
      );

  }
}

// export default AdminConsole;

// import { getBlueprint, getAllVDI } from '../../../redux/server/Blueprint';
// import { getInstances, getConsoles } from '../../../redux/server/Compartment';

function mapDispatchToProps(dispatch) {
    return({
      // signintoserver: (e)=>{dispatch(signupUSER(e))},
      // setthetoken: (e)=>{dispatch(setTOKEN(e))}
      // sendthepost: (e)=>{dispatch(sendPOST(e))},
      // getuserposts: (e)=>{dispatch(userPOSTS(e))}
    })
}

function mapStateToProps(state) {
    return({
      // tokenreturn: state.token,
      // emailreturn: state.email,
      // userpostsreturn: state.retrieveuserposts
    })
}


export default connect(
    mapStateToProps, mapDispatchToProps)(
    About
)
