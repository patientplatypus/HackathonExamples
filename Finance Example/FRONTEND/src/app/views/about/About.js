// @flow weak

import React, {
  PureComponent
}                     from 'react';
import PropTypes      from 'prop-types';
import AnimatedView   from '../../components/animatedView/AnimatedView';

class About extends PureComponent {
  static propTypes= {
    // react-router 4:
    match:    PropTypes.object.isRequired,
    location: PropTypes.object.isRequired,
    history:  PropTypes.object.isRequired,

    // views:
    currentView:  PropTypes.string.isRequired,
    enterAbout:   PropTypes.func.isRequired,
    leaveAbout:   PropTypes.func.isRequired
  };

  componentDidMount() {
    const { enterAbout } = this.props;
    enterAbout();
  }

  componentWillUnmount() {
    const { leaveAbout } = this.props;
    leaveAbout();
  }

  render() {
    return(
      <AnimatedView>
        <h1>
          About
        </h1>
      </AnimatedView>
    );
  }
}

export default About;
