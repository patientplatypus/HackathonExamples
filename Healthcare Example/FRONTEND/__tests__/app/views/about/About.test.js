// @flow weak

import React            from 'react';
import About            from '../../../../src/app/views/about/About';
import renderer         from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { 
  MemoryRouter
}                       from 'react-router';
import {
  shallow,
  mount
}                       from 'enzyme';

describe('About component', () => {
  it('renders as expected', () => {
    const mockProps = {
      // react-router 4:
      match: {},
      location: {},
      history: {},
      // views:
      enterAbout: () => {},
      leaveAbout: () => {},
      currentView: 'About'
    };

    const component = renderer.create(
      <div>
        <MemoryRouter>
          <About {...mockProps} />
        </MemoryRouter>
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('triggers enterAbout on mount', () => {
    const mockEnterAbout = jest.fn();
    const mockProps = {
      // react-router 4:
      match: {},
      location: {},
      history: {},
      // views:
      enterAbout: mockEnterAbout,
      leaveAbout: () => {},
      currentView: 'About'
    };

    /* eslint-disable no-unused-vars */
    // shallow would not call componentDidMount, so we use mount is that case:
    const wrapper = mount(
      <About {...mockProps} />
    );

    // expect(mockEnterAbout).toBeCalled();
    expect(mockEnterAbout.mock.calls.length).toBe(1);
  });

  it('triggers leaveAbout on unMount', () => {
    const mockLeaveAbout = jest.fn();
    const mockProps = {
      // react-router 4:
      match: {},
      location: {},
      history: {},
      // views:
      enterAbout: () => {},
      currentView: 'About',
      leaveAbout: mockLeaveAbout
    };

    const wrapper = shallow(
      <About {...mockProps} />
    );

    wrapper.unmount();
    // expect(mockLeaveAlert).toBeCalled();
    expect(mockLeaveAbout.mock.calls.length).toBe(1);
  });
});
