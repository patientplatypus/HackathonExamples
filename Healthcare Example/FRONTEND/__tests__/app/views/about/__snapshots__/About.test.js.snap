// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component renders as expected 1`] = `
<div>
  <section
    className="content view-enter"
  >
    <h1>
      About
    </h1>
  </section>
</div>
`;
